lab_2_retrieval_w_bm25/main.py

```diff
@@ -207,3 +364,15 @@ def calculate_spearman(rank: list[int], golden_rank: list[int]) -> float | None:
 
     In case of corrupt input arguments, None is returned.
     """
+    is_not_correct = (not rank or not isinstance(rank, list) or
+                      not all(isinstance(i, int) for i in rank) or
+                      not golden_rank or not isinstance(golden_rank, list) or
+                      not all(isinstance(i, int) for i in golden_rank) or
+                      not len(rank) == len(golden_rank))
+    if is_not_correct:
+        return None
+
+    n = len(rank)
+    return 1 - (6 * sum((index - golden_rank.index(number)) ** 2
+                                        for index, number in enumerate(rank)
```

Comment:
It is absolutelly unreadble, move the loop outside the return statement

Reply:
done